src/serialization/compact/SchemaService.ts

```diff
@@ -67,18 +82,26 @@ export class SchemaService {
     /**
      * Puts the schema with the given id to the cluster.
      */
-    put(schema: Schema): Promise<void> {
+    async put(schema: Schema): Promise<void> {
         const schemaId = schema.schemaId;
         const existingSchema = this.schemas.get(schemaId.toString());
         if (existingSchema !== undefined) {
             this.logger.trace('SchemaService', `Schema id ${schemaId} already exists locally`);
             return Promise.resolve();
         }
-        const message = ClientSendSchemaCodec.encodeRequest(schema);
-        const invocation = new Invocation(this.getInvocationService(), message);
-        return this.getInvocationService().invoke(invocation).then(() => {
+
+        const result = await this.replicateSchemaInCluster(schema);
+        if (!result) {
+            throw new IllegalStateError(
+                `The schema ${schema} cannot be replicated in the cluster, after ${this.maxPutRetryCount}  retries. 
```

Comment:
schema.toString does not exist, and this won't look good. What about "the schema with typename schema.typeName"?

Reply:
Fixed